R/generateFilterValues.R

```diff
@@ -125,21 +125,26 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
     }
 
   } else {
-    fval = lapply(filter, function(x) {
-      x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[x$name]]))
-      missing.score = setdiff(fn, names(x))
-      x[missing.score] = NA_real_
-      x[match(fn, names(x))]
-    })
+    index_names = names(method)
+	if (is.null(index_names))
+		index_names = method
+	fval = mapply(function(x, name) {
+			x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[name]]))
+			missing.score = setdiff(fn, names(x))
+			x[missing.score] = NA_real_
+			x[match(fn, names(x))]
+    }, filter, index_names, SIMPLIFY = FALSE)    
     fval = do.call(cbind, fval)
     colnames(fval) = method
-    types = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)
+	if (!is.null(names(method)))
+		colnames(fval) = names(method)
```

Comment:
Would this be obsolete if we would require named inputs right from the start?

Reply:
Yes, but that would break a lot of code. I am trying to make it backwards compatible.

Reply:
Ok fine, same as the comment before then.